src/content/app/tools/blast/components/listed-blast-submission/ListedBlastSubmission.tsx

```diff
@@ -0,0 +1,210 @@
+
+import type { BlastProgram } from 'src/content/app/tools/blast/types/blastSettings';
+
+import styles from './ListedBlastSubmission.scss';
+
+export type Props = {
+  submission: BlastSubmission;
+};
+
+const ListedBlastSubmission = (props: Props) => {
+  const { submission } = props;
+
+  const sequences = submission.submittedData.sequences;
+  const allJobs = submission.results;
+  const isAnyJobRunning = allJobs.some((job) => job.status === 'RUNNING');
+
+  const jobsGroupedBySequence = sequences.map((sequence) => {
+    const jobs = allJobs.filter((job) => job.sequenceId === sequence.id);
+    return {
+      sequence,
+      jobs
+    };
+  });
+
+  const sequenceBoxes = jobsGroupedBySequence.map(({ sequence, jobs }) => (
+    <SequenceBox key={sequence.id} sequence={sequence} jobs={jobs} />
+  ));
+
+  return (
+    <div className={styles.listedBlastSubmission}>
+      <Header {...props} isAnyJobRunning={isAnyJobRunning} />
+      {sequenceBoxes}
+    </div>
+  );
+};
+
+const Header = (
+  props: Props & {
+    isAnyJobRunning: boolean;
+  }
+) => {
+  const { submission } = props;
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const blastProgram =
+    submission.submittedData.parameters.program.toUpperCase();
+  const submissionId = submission.id;
+  const submmissionTime = getDateString(new Date(submission.submittedAt), {
+    withTime: true
+  });
+
+  const editSubmission = () => {
+    const { sequences, species, parameters } = submission.submittedData;
+    const parsedSequences = sequences.flatMap((sequence) =>
+      parseBlastInput(sequence.value)
+    );
+    const { title, program, stype, ...otherParameters } = parameters;
+
+    const payload = {
+      sequences: parsedSequences,
+      selectedSpecies: species,
+      settings: {
+        jobName: title,
+        sequenceType: stype,
+        program: program as BlastProgram,
+        parameters: otherParameters
+      }
+    };
+    dispatch(fillBlastForm(payload));
+    navigate(urlFor.blastForm());
+  };
+
+  const handleDeletion = () => {
+    dispatch(deleteBlastSubmission(submissionId));
+  };
+
+  return (
+    <BlastSubmissionHeaderGrid>
+      <div>{blastProgram}</div>
+      <div>
+        <span className={styles.submissionIdLabel}>Submission</span>
+        <span>{submissionId}</span>
+        <span className={styles.editSubmission} onClick={editSubmission}>
+          Edit/rerun
+        </span>
+        <span className={styles.timeStamp}>
+          <span>{submmissionTime}</span>
+          <span className={styles.timeZone}>GMT</span>
+        </span>
+      </div>
+      <div className={styles.controlButtons}>
+        {!props.isAnyJobRunning && (
+          <>
+            <DeleteButton onClick={handleDeletion} />
+            <DownloadButton className={styles.inactiveButton} />
+          </>
+        )}
+        <ButtonLink
+          to={urlFor.blastSubmission(submission.id)}
```

Comment:
There are two variants of this: `submissionId` and `submission.id`. Can we stick with one?

Reply:
✅